### PR TITLE
add upload code in client side.

### PR DESCRIPTION
**Pull request by byc1234**

### CODE REVIEW

File: znai-client/pom.xml [hunk 1]
```diff
@@ -42,5 +42,33 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpcore</artifactId>
+            <version>4.4.3</version>
+            <scope>compile</scope>
```
Comment (MykolaGolubyev) [inline znai-client/pom.xml:53]: compile is default

File: znai-client/pom.xml [hunk 2]
```diff
@@ -42,5 +42,33 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.5.6</version>
```
Comment (MykolaGolubyev) [inline znai-client/pom.xml:58]: may make sense to define these depenendencies in dep manager and use property to have the same version in all of them?

File: znai-client/pom.xml [hunk 3]
```diff
@@ -42,5 +42,33 @@
...
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
```
Comment (MykolaGolubyev) [inline znai-client/pom.xml:69]: is this the same `httpclient` dependency specified multiple times?

File: znai-client/src/main/java/org/testingisdocumenting/znai/client/upload/DocUploader.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import org.testingisdocumenting.znai.client.ZipTask;
+import org.testingisdocumenting.znai.console.ConsoleOutputs;
+import org.testingisdocumenting.znai.console.ansi.Color;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class DocUploader {
+    private static final String USER = System.getProperty("USER", "");
```
Comment (MykolaGolubyev) [inline znai-client/src/main/java/org/testingisdocumenting/znai/client/upload/DocUploader.java:21]: that is a good version one. I think in the next few versions we should move into using `Principal` on a server side. 
Assuming that principla is going to be set (e.g. nginx in front)

File: znai-enterprise/pom.xml [hunk 5]
```diff
@@ -59,5 +59,11 @@
             <groupId>io.vertx</groupId>
             <artifactId>vertx-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.5.6</version>
```
Comment (MykolaGolubyev) [inline znai-enterprise/pom.xml:65]: same as above, let's move this dep to dependency management section of the parent pom

File: znai-client/pom.xml (hunk 3 above)
Comment (byc1234) [reply znai-client/pom.xml]: Yeah, intellij did hard work to add three of them.

File: pom.xml [hunk 6]
```diff
@@ -356,6 +358,19 @@
...
+                <artifactId>httpcore</artifactId>
+                <version>${httpcore.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.apache.httpcomponents</groupId>
+                <artifactId>httpclient</artifactId>
+                <version>${httpclient.version}</version>
+            </dependency>
+
```
Comment (MykolaGolubyev) [inline pom.xml:373]: could you please remove the space here

File: znai-client/pom.xml [hunk 7]
```diff
@@ -49,26 +49,10 @@
         <dependency>
             <groupId>org.apache.httpcomponents</groupId>
             <artifactId>httpcore</artifactId>
-            <version>4.4.3</version>
-            <scope>compile</scope>
         </dependency>
         <dependency>
```
Comment (MykolaGolubyev) [inline znai-client/pom.xml:53]: could you please add space in between deps?

**Pull request merged**